torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -333,13 +333,15 @@ ProcessGroupGloo::SendWork::SendWork(
 
 bool ProcessGroupGloo::SendWork::wait() {
   bool sendCompleted = false;
-  std::unique_lock<std::mutex> lock(mutex_);
   try {
     sendCompleted = buffer_->waitSend();
   } catch (...) {
+    // Lock to write exception_
+    std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
Let's make them lock guard to be explicit that the following code will not manually unlock them?